File: Tau.KeyVault.PyClient/tau_keyvault/enums.py
"""Enumerations for the Tau Key Vault client."""

from enum import StrEnum


class KeyVaultTransport(StrEnum):
    """Determines how the client communicates with the Tau Key Vault API."""

    API = "Api"
    """Use JSON for all requests and responses (default)."""

    PROTOBUF = "Protobuf"
    """Use Protocol Buffers for all requests and responses."""

    PROTOBUF_WITH_API_FALLBACK = "ProtobufWithApiFallback"
    """Attempt Protobuf first; on failure, retry with JSON automatically."""


class KeyVaultDataType(StrEnum):
    """Data types supported by Tau Key Vault."""

    TEXT = "Text"
    CODE = "Code"
    NUMERIC = "Numeric"
    BOOLEAN = "Boolean"
    DATE = "Date"
    TIME = "Time"
    DATE_TIME = "DateTime"
    JSON = "Json"
    CSV = "Csv"


class KeyVaultAuditAction(StrEnum):
    """Action recorded in the access audit log."""

    READ_KEY = "ReadKey"
    LIST_KEYS = "ListKeys"
    WRITE_KEY = "WriteKey"
    DELETE_KEY = "DeleteKey"
    LIST_ENVIRONMENTS = "ListEnvironments"
    DELETE_ENVIRONMENT = "DeleteEnvironment"
    RENAME_ENVIRONMENT = "RenameEnvironment"
    EXPORT = "Export"
    IMPORT = "Import"
    READ_AUDIT = "ReadAudit"
    AUTH_FAILURE = "AuthFailure"


class KeyVaultAuditOutcome(StrEnum):
    """Result of an audited action."""

    SUCCESS = "Success"
    NOT_FOUND = "NotFound"
    DENIED = "Denied"
    ERROR = "Error"

File: Tau.KeyVault.JsClient/src/enums.js
/**
 * Determines how the client communicates with the Tau Key Vault API.
 * @readonly
 * @enum {string}
 */
export const KeyVaultTransport = Object.freeze({
  /** Use JSON for all requests and responses (default). */
  Api: 'Api',
  /** Use Protocol Buffers for all requests and responses. */
  Protobuf: 'Protobuf',
  /** Attempt Protobuf first; on failure, retry with JSON automatically. */
  ProtobufWithApiFallback: 'ProtobufWithApiFallback',
});

/**
 * Data types supported by Tau Key Vault.
 * @readonly
 * @enum {string}
 */
export const KeyVaultDataType = Object.freeze({
  Text: 'Text',
  Code: 'Code',
  Numeric: 'Numeric',
  Boolean: 'Boolean',
  Date: 'Date',
  Time: 'Time',
  DateTime: 'DateTime',
  Json: 'Json',
  Csv: 'Csv',
});

/** Action recorded in the access audit log. @readonly @enum {string} */
export const KeyVaultAuditAction = Object.freeze({
  ReadKey: 'ReadKey',
  ListKeys: 'ListKeys',
  WriteKey: 'WriteKey',
  DeleteKey: 'DeleteKey',
  ListEnvironments: 'ListEnvironments',
  DeleteEnvironment: 'DeleteEnvironment',
  RenameEnvironment: 'RenameEnvironment',
  Export: 'Export',
  Import: 'Import',
  ReadAudit: 'ReadAudit',
  AuthFailure: 'AuthFailure',
});

/** Result of an audited action. @readonly @enum {string} */
export const KeyVaultAuditOutcome = Object.freeze({
  Success: 'Success',
  NotFound: 'NotFound',
  Denied: 'Denied',
  Error: 'Error',
});
